packages/doenetml-worker-rust/lib-doenetml-core/src/core/dast/macro_resolve.test.rs

```diff
@@ -0,0 +1,247 @@
+use super::*;
+use crate::{
+    dast::{DastIndex, DastText, DastTextMacroContent},
+    test_utils::*,
+};
+
+fn find_node_index_by_name(flat_root: &FlatRoot, name: &str) -> Option<Index> {
+    flat_root.nodes.iter().find_map(|node| match node {
+        FlatNode::Element(e) if e.name == name => Some(e.idx),
+        _ => None,
+    })
+}
+
+fn make_path<'a, T: AsRef<[&'a str]>>(path_str: T) -> Vec<PathPart> {
+    let path_str = path_str.as_ref();
+    path_str
+        .iter()
+        .map(|s| PathPart {
+            name: s.to_string(),
+            index: Vec::new(),
+            position: None,
+        })
+        .collect()
+}
+
+#[test]
+fn can_resolve_name_among_parents() {
+    let dast_root = dast_root_no_position(
+        r#"<a name="x">
+            <b name="y">
+                <c name="z" />
+            </b>
+        </a>
+        <d name="y" />"#,
+    );
+    let flat_root = FlatRoot::from_dast(&dast_root);
+    let b_idx = find_node_index_by_name(&flat_root, "b").unwrap();
```

Comment:
The terminology clashes for "name" and "type" in DoenetML versus the dast is hurting here. This line is confusing since the above DoenetML make it seem that the names are "x", "y" and "z". In DoenetML, "name" become "component type". Since the DoenetML terminology is more front and center in the context of this test, maybe `find_node_index_by_component_type()`?

Reply:
How about I just change it to `find`?